mesonbuild/mparser.py

```diff
@@ -194,14 +193,11 @@ def lex(self, filename: str) -> T.Generator[Token, None, None]:
                     elif tid in {'string', 'fstring'}:
                         # Handle here and not on the regexp to give a better error message.
                         if match_text.find("\n") != -1:
-                            mlog.warning(textwrap.dedent("""\
-                                    Newline character in a string detected, use ''' (three single quotes) for multiline strings instead.
-                                    This will become a hard error in a future Meson release.\
-                                """),
-                                self.getline(line_start),
-                                str(lineno),
-                                str(col)
-                            )
+                            msg = ParseException("Newline character in a string detected, use ''' (three single quotes) "
+                                                 "for multiline strings instead.\n"
+                                                 "This will become a hard error in a future Meson release.",
+                                                 self.getline(line_start), lineno, col)
+                            mlog.warning(msg, location=BaseNode(lineno, col, filename))
```

Comment:
I find the triple quoted, demented version much easier to read, since it looks in the editor exactly his it will be printed

Reply:
And yet it results in flake8 indentation warnings, junk whitespace at the end, and accomplishes nothing because it isn't being printed any which way in particular, the exact formatted layout is not meaningful.

I'm basically pretty meh on that.

Reply:
(To be clear: flake8 complains because it cannot figure out a good way to indent the line immediately after, because apparently `func1(func2('string'), otherarg)` cannot be happy when string is multiline. Go figure.)